ert_data/measured.py

```diff
@@ -1,154 +1,196 @@
+except ImportError:
+    from pkg_resources import get_distribution
+    __version__ = get_distribution("ert").version
+
 
-class MeasuredData(object):
-    def __init__(self, facade, keys, index_lists=None, load_data=True):
-        self._facade = facade
-        self._set_data(self._get_data(keys, index_lists, load_data))
+class MeasuredData(pd.DataFrame):
+    def __init__(self, facade, keys, index_lists=None, load_data=True, case_name=None):
+        """
+        This is a bit ugly, but is close to the way pandas does this internally.
+        The problem is that when an operation is performed on a DataFrame, a view of
+        the dataframe is returned. We want that returned dataframe to be a MeasuredData
+        object, but we dont want to read from libres each time, so we need to have a
+        variable constructor.
+        """
+        if isinstance(facade, pd.core.internals.BlockManager):
+            super().__init__(facade)
+        else:
+            if not keys:
+                raise loader.ObservationError("No observation keys provided")
+            if case_name is None:
+                case_name = facade.get_current_case_name()
+            df = _get_data(facade, keys, load_data, case_name)
+            if index_lists:
+                df = MeasuredData.filter_on_column_index(df, keys, index_lists)
+            super().__init__(data=df.values, columns=df.columns, index=df.index)
 
     @property
-    def data(self):
-        return self._data
+    def _constructor(self):
+        """
+        Part of the pandas API
+        """
+        return MeasuredData._internal_ctor
 
-    def _set_data(self, data):
-        expected_keys = ["OBS", "STD"]
-        if not isinstance(data, pd.DataFrame):
-            raise TypeError(
-                "Invalid type: {}, should be type: {}".format(type(data), pd.DataFrame)
-            )
-        elif not set(expected_keys).issubset(data.index):
-            raise ValueError(
-                "{} should be present in DataFrame index, missing: {}".format(
-                    ["OBS", "STD"], set(expected_keys) - set(data.index)
-                )
-            )
-        else:
-            self._data = data
+    @classmethod
+    def _internal_ctor(cls, *args, **kwargs):
+        """
+        This is a part of creating the constructor when pandas returns a subframe
+        we just insert None as the keys (it can be anything as we dont use it)
+        """
+        kwargs['keys'] = None
+        return cls(*args, **kwargs)
 
-    def remove_failed_realizations(self):
-        self._set_data(self._remove_failed_realizations())
+    @property
+    def response(self):
+        return self._get_simulated_data()
 
+    @property
+    def observations(self):
+        return self.loc[["OBS", "STD"]]
+
+    @property
+    def data(self):
+        return self
+
+    def remove_failed_realizations(self):
+        return self._remove_failed_realizations()
+
+    @deprecation.deprecated(
```

Comment:
It is possible to also add the `@deprecation.fail_if_not_removed` decorator on tests that are calling these functions in order to get a failing test if we forget to remove the methods.